functions/snippets/src/main/java/com/example/functions/OcrProcessImage.java

```diff
@@ -0,0 +1,152 @@
+  private static Publisher publisher;
+  private static final Gson gson = new Gson();
+
+  public OcrProcessImage() throws IOException {
+    publisher = Publisher.newBuilder(
+        ProjectTopicName.of(PROJECT_ID, TRANSLATE_TOPIC_NAME)).build();
+  }
+
+  @Override
+  public void accept(GcsEvent gcsEvent, Context context) throws RuntimeException {
+
+    // Validate parameters
+    String bucket = gcsEvent.getBucket();
+    if (bucket == null) {
+      throw new IllegalArgumentException("Missing bucket parameter");
+    }
+    String filename = gcsEvent.getName();
+    if (filename == null) {
+      throw new IllegalArgumentException("Missing name parameter");
+    }
+
+    detectText(bucket, filename);
+  }
+  // [END functions_ocr_process]
+
+  // [START functions_ocr_detect]
+  private void detectText(String bucket, String filename) throws RuntimeException {
+    LOGGER.info("Looking for text in image " + filename);
+
+    List<AnnotateImageRequest> visionRequests = new ArrayList<>();
+    String gcsPath = String.format("gs://%s/%s", bucket, filename);
+
+    ImageSource imgSource = ImageSource.newBuilder().setGcsImageUri(gcsPath).build();
+    Image img = Image.newBuilder().setSource(imgSource).build();
+
+    Feature textFeature = Feature.newBuilder().setType(Feature.Type.TEXT_DETECTION).build();
+    AnnotateImageRequest visionRequest =
+        AnnotateImageRequest.newBuilder().addFeatures(textFeature).setImage(img).build();
+    visionRequests.add(visionRequest);
+
+    // Detect text in an image using the Cloud Vision API
+    AnnotateImageResponse visionResponse;
+    try (ImageAnnotatorClient client = ImageAnnotatorClient.create()) {
+      visionResponse = client.batchAnnotateImages(visionRequests).getResponses(0);
+      if (visionResponse.hasError()) {
+        LOGGER.severe("Error detecting text: " + visionResponse.getError().getMessage());
+        return;
+      }
+    } catch (IOException e) {
+      // Cast to RuntimeException
+      throw new RuntimeException(e);
+    }
+
+    if (visionResponse == null || !visionResponse.hasFullTextAnnotation()) {
+      return;
+    }
+
+    String text = visionResponse.getFullTextAnnotation().getText();
+    LOGGER.info("Extracted text from image: " + text);
+
+    // Detect language using the Cloud Translation API
+    DetectLanguageRequest languageRequest =
+        DetectLanguageRequest.newBuilder()
+            .setParent(LOCATION_NAME.toString())
+            .setMimeType("text/plain")
+            .setContent(text)
+            .build();
+    DetectLanguageResponse languageResponse;
+    try (TranslationServiceClient client = TranslationServiceClient.create()) {
+      languageResponse = client.detectLanguage(languageRequest);
+    } catch (IOException e) {
+      // Cast to RuntimeException
+      throw new RuntimeException(e);
+    }
+
+    if (languageResponse == null || languageResponse.getLanguagesCount() == 0) {
+      return;
```

Comment:
Is this a success?

Reply:
Added a log statement. (See above comment for more info.)